pkg/storage/store.go

```diff
@@ -3096,6 +3096,7 @@ func (s *Store) processRaftRequest(
 		if req.Message.Type == raftpb.MsgApp {
 			r.setEstimatedCommitIndexLocked(req.Message.Commit)
 		}
+		r.setLastActivityLocked(req.FromReplica.ReplicaID)
```

Comment:
Is it possible for a hopelessly lagged replica to come back online and require a snapshot, but still report its log index and cause the quota pool to be immediately drained to a negative value?